// THIS FILE IS AUTOMATICALLY GENERATED by https://github.com/noppoMan/aws-sdk-swift/blob/master/Sources/CodeGenerator/main.swift. DO NOT EDIT.

import AWSSDKSwiftCore

/// Error enum for AppSync
public enum AppSyncErrorType: AWSErrorType {
    case badRequestException(message: String?)
    case notFoundException(message: String?)
    case limitExceededException(message: String?)
    case unauthorizedException(message: String?)
    case internalFailureException(message: String?)
    case apiKeyLimitExceededException(message: String?)
    case apiKeyValidityOutOfBoundsException(message: String?)
    case concurrentModificationException(message: String?)
    case apiLimitExceededException(message: String?)
    case graphQLSchemaException(message: String?)
}

extension AppSyncErrorType {
    public init?(errorCode: String, message: String?){
        var errorCode = errorCode
        if let index = errorCode.firstIndex(of: "#") {
            errorCode = String(errorCode[errorCode.index(index, offsetBy: 1)...])
        }
        switch errorCode {
        case "BadRequestException":
            self = .badRequestException(message: message)
        case "NotFoundException":
            self = .notFoundException(message: message)
        case "LimitExceededException":
            self = .limitExceededException(message: message)
        case "UnauthorizedException":
            self = .unauthorizedException(message: message)
        case "InternalFailureException":
            self = .internalFailureException(message: message)
        case "ApiKeyLimitExceededException":
            self = .apiKeyLimitExceededException(message: message)
        case "ApiKeyValidityOutOfBoundsException":
            self = .apiKeyValidityOutOfBoundsException(message: message)
        case "ConcurrentModificationException":
            self = .concurrentModificationException(message: message)
        case "ApiLimitExceededException":
            self = .apiLimitExceededException(message: message)
        case "GraphQLSchemaException":
            self = .graphQLSchemaException(message: message)
        default:
            return nil
        }
    }
}
